Unix/installbuilder/datafiles/Linux.data

```diff
@@ -189,6 +190,40 @@ ConfigureOmiService() {
     ${{SERVICE_CTL}} start
 }
 
+ConfigureLogRotate()
+{
+    echo "Configure log rotate for OMI"
+    # create the logrotate file if it doesn't exist
+    if [ ! -f /etc/logrotate.d/omi ]; then
+        cat /etc/opt/omi/conf/omilogrotate.conf > /etc/logrotate.d/omi
+    fi
+}
+
+ConfigureCronForLogRotate()
+{
+    echo "Checking if cron is installed..."
+    # warn user that he need to install cron if cron doesn't install
+    crontab -l >/dev/null 2>&1
+    if [ $? -ne 0 ]; then
+        echo "WARNING: LogRotate can't be enabled, please install cron at first!"
+        return
+    fi
+
+    echo "Checking if cron/crond service is started..."
+    # warn user that he need to start cron/crond service if cron doesn't start
+    cronid=$(pidof cron > /dev/null 2>&1)
+    crondid=$(pidof crond > /dev/null 2>&1)
+    if [ ! -z "$cronid" -a ! -z "$crondid" ]; then
+        echo "WARNING: LogRotate can be enabled, but please start cron/crond service!"
+    fi
+
+    echo "Set up a cron job to OMI logrotate every 15 minutes"
+    # create the cron file if it doesn't exist
+    if [ ! -f /etc/cron.d/omilogrotate ]; then
+        (echo "*/15 * * * * root /usr/sbin/logrotate /etc/logrotate.d/omi >/dev/null 2>&1" > /etc/cron.d/omilogrotate) > /dev/null 2>&1
+    fi
+}
```

Comment:
The mechanism that Bruce used to add Kerberos keytab file entry in cron is different than how you are doing it.  Could we all do it the same way?

I'm also worried that not all systems will look in /etc/cron.d/*.

Reply:
BTW if `/etc/cron.d` and `/etc/logrotate.d` don't exist, then that means the system probably don't enable crontab or don't install crontab, and we will not meet error while installing in the situation and we just not enable our log rotate on it since it is not installed or not supported.

Reply:
From what I can research, /etc/cron.d is only used on Debian systems.  Please verify that this works on non-Debian systems also.

Reply:
you can check here: https://github.com/Microsoft/OMS-Agent-for-Linux/blob/aa3973a65343eb4b4e557b97766654765a34cc0f/docs/Troubleshooting.md#on-ubuntudebian , it works on all Linux, thanks.